tools/iree-prof-output-utils.cc

```diff
@@ -22,8 +23,60 @@ const char* GetZoneName(const tracy::Worker& worker,
   return worker.GetString(srcloc.name.active ? srcloc.name : srcloc.function);
 }
 
-const char* GetThreadName(const tracy::Worker& worker, uint16_t tid) {
-  return worker.GetThreadName(worker.DecompressThread(tid));
+void MergeDuration(int64_t start, int64_t end, std::vector<int64_t>& merged) {
+  CHECK_LE(start, end);
+  if (start == end) {
+    return;  // No changes.
+  }
+
+  int least_index_greater_than_or_equal_to_start = 0;
+  while (least_index_greater_than_or_equal_to_start < merged.size() &&
+         start > merged[least_index_greater_than_or_equal_to_start]) {
+    ++least_index_greater_than_or_equal_to_start;
+  }
+
+  int greatest_index_less_than_or_equal_to_end = merged.size() - 1;
+  while (greatest_index_less_than_or_equal_to_end >= 0 &&
+         end < merged[greatest_index_less_than_or_equal_to_end]) {
+    --greatest_index_less_than_or_equal_to_end;
+  }
+
+  // least_index_greater_than_or_equal_to_start can be at most
+  // greatest_index_less_than_or_equal_to_end + 1.
+  CHECK_LE(least_index_greater_than_or_equal_to_start,
+           greatest_index_less_than_or_equal_to_end + 1);
+
+  bool start_is_within_an_existing_duration =
+      least_index_greater_than_or_equal_to_start % 2 != 0;
+  bool end_is_within_an_existing_duration =
+      greatest_index_less_than_or_equal_to_end % 2 == 0;
+
+  // Merge all durations between start, end into one duration
+  merged.erase(merged.begin() + least_index_greater_than_or_equal_to_start,
```

Comment:
nit: The function can be expensive to run depending on the input's characteristic. If z.second.zones is already sorted in the time, we might not really touch the items at the end, but if it is not, the erase and insert in the middle of vector can be very expensive.

It reminds me of this coding questions:
https://leetcode.com/problems/merge-intervals/solutions/4519554/step-by-step-explanation-easy-to-understand-in-cpp-javal-js-python/

Reply:
Changed to use only zone in the timeline of thread data which reduced most of expensive operations.